test/test_typing.py

```diff
@@ -0,0 +1,150 @@
+# based on NumPy numpy/typing/tests/test_typing.py
+
+import itertools
+import os
+import re
+import shutil
+from typing import IO, Dict, List
+
+import pytest
```

Comment:
Please remove `pytest` dependency (as I've mentioned in my previous comment, pytorch test infra is not compatible with it yet)

Reply:
Oops, my bad. Do you know if pytest support is landing in a near future?

Reply:
as far as i know there's no plan. see: https://github.com/pytorch/pytorch/blob/master/CONTRIBUTING.md#unit-testing

Reply:
> Please remove `pytest` dependency (as I've mentioned in my previous comment, pytorch test infra is not compatible with it yet)

After looking into it, this isn't quite right - there are unconditional `import pytest` statements in the codebase (see 
https://github.com/pytorch/pytorch/issues/11578#issuecomment-787485279). There's no point spending significant effort rewriting this bit of by now well-tested code. I think the correct solution is to keep this file as is, and:
- add the file name to `USE_PYTEST_LIST` in `test/run_test.py`
- make the test runnable standalone by, at the end of the file, adding:
```
if __name__ == '__main__':
    pytest.main([__file__])
```

Reply:
Thanks for the suggestion Ralf. I've put the file on `USE_PYTEST_LIST`